packages/wonder-blocks-dropdown/src/components/select-opener.tsx

```diff
@@ -97,74 +131,52 @@ export default class SelectOpener extends React.Component<SelectOpenerProps> {
-                        light,
-                        isPlaceholder,
-                        error,
-                    );
-                    const {hovered, focused, pressed} = state;
-
-                    // The icon colors are kind of fickle. This is just logic
-                    // based on the zeplin design.
-                    const iconColor = light
-                        ? disabled || pressed
-                            ? "currentColor"
-                            : tokens.color.white
-                        : disabled
-                        ? tokens.color.offBlack32
-                        : tokens.color.offBlack64;
+        const stateStyles = _generateStyles(light, isPlaceholder, error);
 
-                    const style = [
-                        styles.shared,
-                        stateStyles.default,
-                        disabled && stateStyles.disabled,
-                        !disabled &&
-                            (pressed
-                                ? stateStyles.active
-                                : (hovered || focused) && stateStyles.focus),
-                    ];
+        // The icon colors are kind of fickle. This is just logic
+        // based on the zeplin design.
+        const iconColor = light
+            ? disabled || error
+                ? "currentColor"
+                : tokens.color.white
+            : disabled
+            ? tokens.color.offBlack32
+            : tokens.color.offBlack64;
 
-                    return (
-                        <StyledButton
-                            {...sharedProps}
-                            aria-disabled={disabled}
-                            aria-expanded={open ? "true" : "false"}
-                            aria-haspopup="listbox"
-                            data-testid={testId}
-                            id={id}
-                            style={style}
-                            type="button"
-                            {...childrenProps}
-                        >
-                            <LabelMedium style={styles.text}>
-                                {/* Note(tamarab): Prevents unwanted vertical
-                                shift for empty selection */}
-                                {children || "\u00A0"}
-                            </LabelMedium>
-                            <PhosphorIcon
-                                icon={caretDownIcon}
-                                color={iconColor}
-                                size="small"
-                                style={styles.caret}
-                                aria-hidden="true"
-                            />
-                        </StyledButton>
-                    );
-                }}
-            </ClickableBehavior>
-        );
-    }
+        const style = [
+            styles.shared,
+            stateStyles.default,
+            disabled && stateStyles.disabled,
+            this.state.pressed && stateStyles.pressed,
```

Comment:
**suggestion(change-required):** This also needs to check that the opener is not `disabled`, otherwise this will display the pressed/active state when the control is disabled and the user presses `Enter/Space`.